cps/editbooks.py

```diff
@@ -332,7 +332,8 @@ def request_media_download(media_url, original_url):
     
     if request.method == "POST" and "mediaURL" in request.form:
         media_url = request.form["mediaURL"]
-        original_url = request.referrer + "meta"
+        original_url = request.form["serverURL"]
+        original_url = original_url.replace("/media", "/meta")
```

Comment:
Is this a REGEX that should possible be a bit safer, e.g. in cases where implementer chose http://box.lan/media in their school?  (e.g. to ensure http://box.lan/media/media is being changed to http://box.lan/media/meta ?)

```suggestion
        original_url = original_url.replace("/media$", "/meta")
```

Reply:
Thanks for mentioning this particular use case. Let's make sure that only occurrence of "/media" at the end of the URL path is replaced with "/meta".